src/Widgets/SettingsPopover.vala

```diff
@@ -113,6 +113,11 @@ public sealed class Terminal.SettingsPopover : Gtk.Popover {
             active = Application.settings.get_boolean ("natural-copy-paste")
         };
 
+        var unsafe_paste_alert_button = new Granite.SwitchModelButton (_("Unsafe Paste Alert")) {
+            description = _("Show a warning dialog when pasting a command with 'sudo' or multiple lines"),
```

Comment:
maybe something like this? I don’t think we need to say “dialog”. Since there’s multiple ways to enter text from somewhere else, maybe we focus on the copying instead of the pasting/dragging. And the kind of person who needs this probably doesn’t understand what “sudo” or “doas” means so try to use language they would understand. Not sure if “administrative” or “advanced” or something else is better here. Might need to look at language from like Polkit agent or something else to see what we usually use here 

```suggestion
            description = _("Warn when copied text uses advanced permissions or contains multiple commands"),
```

Reply:
Agree about "dialog". "Copied text" is a bit ambiguous as you can also copy *from* the terminal.  Again not sure what "advanced" means in this context (and what it might be translated to).  It is difficult to be precise, unambiguous and succinct all at the same time!

Reply:
Okay updated my suggestion to use "administrative commands" and stick to "paste" here

Reply:
"Warn when pasted text contains multiple or administrative commands"

short. May be of the vague.

Reply:
I think that is good - short and clear!

Reply:
I think that's pretty good. In combination with seeing/interacting with the actual warning dialog I think folks should be able to work out what this setting refers to

Reply:
Yes. By default it is "on". So if they disable it, theyll likely have already interacted with it in the past and connected the dots with the dialog and pasting "doas"

(unrelated but does it also react to Wget outputted to bash ? Ive seen too "oh heres a one liner to download and run this script that fixes your problem")

Reply:
At present it just looks for the words "sudo" or "doas" so it is pretty dumb. Maybe raise an issue if you feel there is a further security threat not covered.